guidance/d_star_lite/d_star_lite/d_star_lite.py

```diff
@@ -0,0 +1,464 @@
+            next_point (DSLNode): The next point.
+        """
+        if not self.WP:  # If the waypoint list is empty
+            self.WP.append(current_point)
+        else:
+            # Get the last waypoint
+            last_wp = self.WP[-1]
+            # Determine directions
+            last_direction = self.get_direction(last_wp, current_point)
+            current_direction = self.get_direction(current_point, next_point)
+            
+            # If there's a change in direction, add the current point to waypoints
+            if current_direction != last_direction:
+                #print("Change in direction detected")
+                self.WP.append(current_point)
+
+    def compare_keys(self, key_pair1: tuple[float, float], key_pair2: tuple[float, float]) -> bool:
+        """
+        Compares the priority keys of two nodes.
+
+        Args:
+            key_pair1 (tuple): The first key pair to compare.
+            key_pair2 (tuple): The second key pair to compare.
+        
+        Returns:
+            bool: True if `key_pair1` should precede `key_pair2` in sorting order, False otherwise.
+        """
+        return key_pair1[0] < key_pair2[0] or (key_pair1[0] == key_pair2[0] and key_pair1[1] < key_pair2[1])
+    
+    def compute_shortest_path(self):
+        """
+        Computes or recomputes the shortest path from the start to the goal using the D* Lite algorithm.
+
+        This method iteratively updates the priorities and costs of nodes based on the graph's current state,
+        adjusting the path as necessary until the start DSLNode's key does not precede the smallest key in the
+        priority queue and the start DSLNode's rhs and g values are equal.
+        """
+        self.U.sort(key=lambda x: x[1])
+        has_elements = len(self.U) > 0
+        start_key_not_updated = self.compare_keys(self.U[0][1], self.calculate_key(self.start))
+        rhs_not_equal_to_g = self.rhs[self.start.x][self.start.y] != self.g[self.start.x][self.start.y]
+        while has_elements and start_key_not_updated or rhs_not_equal_to_g:
+            self.kold = self.U[0][1]
+            u = self.U[0][0]
+            self.U.pop(0)
+            if self.compare_keys(self.kold, self.calculate_key(u)):
+                self.U.append((u, self.calculate_key(u)))
+                self.U.sort(key=lambda x: x[1])
+            elif (self.g[u.x][u.y] > self.rhs[u.x, u.y]).any():
+                self.g[u.x, u.y] = self.rhs[u.x, u.y]
+                for s in self.pred(u):
+                    self.update_vertex(s)
+            else:
+                self.g[u.x, u.y] = math.inf
+                for s in self.pred(u) + [u]:
+                    self.update_vertex(s)
+            self.U.sort(key=lambda x: x[1])
+            start_key_not_updated = self.compare_keys(self.U[0][1], self.calculate_key(self.start))
+            rhs_not_equal_to_g = self.rhs[self.start.x][self.start.y] != self.g[self.start.x][self.start.y]
```

Comment:
This block of code is not very readable, try to add some comments aswell as spacing between if-statements